### ISSUE TITLE
Unicode Decode error on windows with tox 4.0

### ISSUE THREAD

**Issue by maxnoe** [tox-dev/tox#2632]
## Issue

We now get a "UnicodeDecodeError" invalid utf-8 in our windows CI runs with tox 4.0:
https://github.com/astropy/astropy/actions/runs/3642365441/jobs/6149393998

## Environment

Provide at least:

- OS: windows on github CI
- See console output of linked CI run

## Output of running tox

```console
Run python -m tox -e py39-test-alldeps  -- --cov-report=xml:${GITHUB_WORKSPACE}\\coverage.xml  --durations=50
  python -m tox -e py39-test-alldeps  -- --cov-report=xml:${GITHUB_WORKSPACE}\\coverage.xml  --durations=50
  shell: C:\Program Files\Git\bin\sh.EXE -e {0}
  env:
    pythonLocation: C:\hostedtoolcache\windows\Python\3.9.13\x64
    PKG_CONFIG_PATH: C:\hostedtoolcache\windows\Python\3.9.13\x64/lib/pkgconfig
    Python_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.9.13\x64
    Python2_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.9.13\x64
    Python3_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.9.13\x64
py39-test-alldeps: install_deps> python -I -m pip install pytest-openfiles==0.4.0
.pkg: install_requires> python -I -m pip install cython==0.29.30 extension-helpers oldest-supported-numpy setuptools setuptools_scm>=6.2 wheel
.pkg: _optional_hooks> python C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_sdist> python C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta
.pkg: prepare_metadata_for_build_wheel> python C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta
.pkg: build_sdist> python C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta
py39-test-alldeps: install_package_deps> python -I -m pip install PyYAML>=3.13 asdf>=2.10.0 beautifulsoup4 bleach bottleneck certifi coverage[toml] dask[array] fsspec[http]>=2022.8.2 h5py html5lib ipython>=4.2 jplephem matplotlib!=3.4.0,!=3.5.2,>=3.1 mpmath numpy>=1.20 objgraph packaging>=19.0 pandas pooch pyarrow>=5.0.0 pyerfa>=2.0 pytest-astropy-header>=0.2.1 pytest-astropy>=0.10 pytest-doctestplus>=0.12 pytest-xdist pytest>=7.0 pytz s3fs>=2022.8.2 scipy>=1.5 sgp4>=2.3 skyfield>=1.20 sortedcontainers typing-extensions>=3.10.0.1
py39-test-alldeps: install_package> python -I -m pip install --force-reinstall --no-deps D:\a\astropy\astropy\.tox\.pkg\dist\astropy-5.3.dev87+g035338871.tar.gz
py39-test-alldeps: commands[0] .tempdir> pip freeze
aiobotocore==2.4.1
aiohttp==3.8.3
aioitertools==0.11.0
aiosignal==1.3.1
appdirs==1.4.4
asdf==2.14.2
asdf-standard==1.0.3
asdf-transform-schemas==0.3.0
asdf-unit-schemas==0.1.0
astropy @ file:///D:/a/astropy/astropy/.tox/.pkg/dist/astropy-5.3.dev87%2Bg035338871.tar.gz
asttokens==2.2.1
async-timeout==4.0.2
attrs==22.1.0
backcall==0.2.0
beautifulsoup4==4.11.1
bleach==5.0.1
botocore==1.27.59
Bottleneck==1.3.5
certifi==2022.9.24
charset-normalizer==2.1.1
click==8.1.3
cloudpickle==2.2.0
colorama==0.4.6
contourpy==1.0.6
coverage==6.5.0
cycler==0.11.0
dask==2022.12.0
decorator==5.1.1
exceptiongroup==1.0.4
execnet==1.9.0
executing==1.2.0
fonttools==4.38.0
frozenlist==1.3.3
fsspec==2022.11.0
graphviz==0.20.1
h5py==3.7.0
html5lib==1.1
hypothesis==6.60.0
idna==3.4
importlib-metadata==5.1.0
iniconfig==1.1.1
ipython==8.7.0
jedi==0.18.2
jmespath==1.0.1
jplephem==2.18
jsonschema==4.17.3
kiwisolver==1.4.4
locket==1.0.0
matplotlib==3.6.2
matplotlib-inline==0.1.6
mpmath==1.2.1
multidict==6.0.3
numpy==1.23.5
objgraph==3.5.0
packaging==21.3
pandas==1.5.2
parso==0.8.3
partd==1.3.0
pickleshare==0.7.5
Pillow==9.3.0
pluggy==1.0.0
pooch==1.6.0
prompt-toolkit==3.0.36
psutil==5.9.4
pure-eval==0.2.2
pyarrow==10.0.1
pyerfa==2.0.0.1
Pygments==2.13.0
pyparsing==3.0.9
pyrsistent==0.19.2
pytest==7.2.0
pytest-arraydiff==0.5.0
pytest-astropy==0.10.0
pytest-astropy-header==0.2.2
pytest-cov==4.0.0
pytest-doctestplus==0.12.1
pytest-filter-subpackage==0.1.1
pytest-mock==3.10.0
pytest-openfiles==0.4.0
pytest-remotedata==0.3.3
pytest-xdist==3.1.0
python-dateutil==2.8.2
pytz==2022.6
PyYAML==6.0
requests==2.28.1
s3fs==2022.11.0
scipy==1.9.3
semantic-version==2.10.0
sgp4==2.21
six==1.16.0
skyfield==1.45
sortedcontainers==2.4.0
soupsieve==2.3.2.post1
stack-data==0.6.2
tomli==2.0.1
toolz==0.12.0
traitlets==5.6.0
typing_extensions==4.4.0
urllib3==1.26.13
wcwidth==0.2.5
webencodings==0.5.1
wrapt==1.14.1
yarl==1.8.2
zipp==3.11.0
py39-test-alldeps: commands[1] .tempdir> pytest --pyargs astropy D:\a\astropy\astropy/docs --cov-report=xml:D:\a\astropy\astropy\coverage.xml --durations=50
============================= test session starts =============================
platform win32 -- Python 3.9.13, pytest-7.2.0, pluggy-1.0.0
cachedir: .tox\py39-test-alldeps\.pytest_cache

Running tests with Astropy version 5.3.dev87+g035338871.
Running tests in astropy D:\a\astropy\astropy/docs.

Date: 2022-12-07T20:04:58

Platform: Windows-10-10.0.20348-SP0

Executable: D:\a\astropy\astropy\.tox\py39-test-alldeps\Scripts\python.EXE

Full Python Version: 
3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]

encodings: sys: utf-8, locale: cp1252, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.23.5
Scipy: 1.9.3
Matplotlib: 3.6.2
h5py: 3.7.0
Pandas: 1.5.2
PyERFA: 2.0.0.1
Cython: not available
Scikit-image: not available
asdf: 2.14.2
pyarrow: 10.0.1

Using Astropy options: remote_data: none.

ARCH_ON_CI: undefined
IS_CRON: undefined

rootdir: D:\a\astropy\astropy, configfile: setup.cfg
plugins: asdf-2.14.2, hypothesis-6.60.0, arraydiff-0.5.0, astropy-0.10.0, astropy-header-0.2.2, cov-4.0.0, doctestplus-0.12.1, filter-subpackage-0.1.1, mock-3.10.0, openfiles-0.4.0, remotedata-0.3.3, xdist-3.1.0
collected 24843 items
```

LOTS OF TEST OUTPUT

```
============================ slowest 50 durations =============================
6.11s call     .tox/py39-test-alldeps/lib/site-packages/astropy/samp/tests/test_web_profile.py::TestWebProfile::test_web_profile
5.08s teardown .tox/py39-test-alldeps/lib/site-packages/astropy/samp/tests/test_web_profile.py::TestWebProfile::test_web_profile
4.54s call     .tox/py39-test-alldeps/lib/site-packages/astropy/coordinates/tests/test_angles.py::test_angle_multithreading
2.80s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model18]
2.29s call     .tox/py39-test-alldeps/lib/site-packages/astropy/nddata/tests/test_nduncertainty.py::test_for_leak_with_uncertainty
2.20s call     .tox/py39-test-alldeps/lib/site-packages/astropy/utils/tests/test_console.py::test_progress_bar_map
2.18s call     .tox/py39-test-alldeps/lib/site-packages/astropy/time/tests/test_precision.py::test_datetime_timedelta_roundtrip[utc]
2.00s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/fits/tests/test_image.py::TestImageFunctions::test_open_scaled_in_update_mode
2.00s teardown .tox/py39-test-alldeps/lib/site-packages/astropy/samp/tests/test_web_profile.py::TestWebProfile::test_main
1.91s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model92]
1.91s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model17]
1.85s call     .tox/py39-test-alldeps/lib/site-packages/astropy/convolution/tests/test_convolve_fft.py::TestConvolve2D::test_padding
1.78s call     .tox/py39-test-alldeps/lib/site-packages/astropy/utils/tests/test_data.py::test_download_parallel_update
1.71s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model19]
1.62s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/fits/tests/test_image.py::TestCompressedImage::test_rewriting_large_scaled_image_compressed
1.60s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/fits/tests/test_hdulist.py::TestHDUListFunctions::test_hdul_fromstring
1.57s call     .tox/py39-test-alldeps/lib/site-packages/astropy/time/tests/test_precision.py::test_datetime_to_timedelta[utc]
1.53s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/tests/test_parquet.py::test_parquet_mixins_read_one_name[QTable]
1.49s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/fits/tests/test_table.py::TestTableFunctions::test_reference_leak
1.49s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model80]
1.46s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model11]
1.40s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/tests/test_parquet.py::test_parquet_mixins_read_one_name[Table]
1.36s call     .tox/py39-test-alldeps/lib/site-packages/astropy/modeling/tests/test_constraints.py::test_2d_model[LevMarLSQFitter0]
1.36s call     .tox/py39-test-alldeps/lib/site-packages/astropy/modeling/tests/test_constraints.py::test_2d_model[LevMarLSQFitter1]
1.34s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model40]
1.32s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model82]
1.32s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model20]
1.29s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model27]
1.27s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model24]
1.26s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model67]
1.25s setup    .tox/py39-test-alldeps/lib/site-packages/astropy/coordinates/tests/test_angles.py::test_str_repr_angles_nan[input0-nan-nan deg-Longitude]
1.24s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model64]
1.23s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model46]
1.21s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model45]
1.20s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model63]
1.18s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model89]
1.16s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model21]
1.13s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model9]
1.12s call     .tox/py39-test-alldeps/lib/site-packages/astropy/time/tests/test_precision.py::test_sidereal_lat_independent[apparent]
1.12s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model22]
1.12s call     .tox/py39-test-alldeps/lib/site-packages/astropy/time/tests/test_precision.py::test_sidereal_lat_independent[mean]
1.10s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model74]
1.08s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model66]
1.04s call     .tox/py39-test-alldeps/lib/site-packages/astropy/time/tests/test_precision.py::test_sidereal_lon_independent[apparent]
1.04s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model36]
1.04s call     .tox/py39-test-alldeps/lib/site-packages/astropy/timeseries/periodograms/lombscargle/tests/test_statistics.py::test_inverse_bootstrap[True-False-log]
1.03s call     .tox/py39-test-alldeps/lib/site-packages/astropy/timeseries/periodograms/lombscargle/tests/test_statistics.py::test_inverse_bootstrap[True-True-log]
1.03s call     .tox/py39-test-alldeps/lib/site-packages/astropy/timeseries/periodograms/lombscargle/tests/test_statistics.py::test_inverse_bootstrap[True-False-psd]
1.03s call     .tox/py39-test-alldeps/lib/site-packages/astropy/io/misc/asdf/tags/transform/tests/test_transform.py::test_single_model[model86]
1.03s call     .tox/py39-test-alldeps/lib/site-packages/astropy/timeseries/periodograms/lombscargle/tests/test_statistics.py::test_inverse_bootstrap[True-True-model]
========= 24429 passed, 281 skipped, 133 xfailed in 518.77s (0:08:38) =========
py39-test-alldeps: internal error
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\session\cmd\run\single.py", line 46, in _evaluate
    code, outcomes = run_commands(tox_env, no_test)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\session\cmd\run\single.py", line 81, in run_commands
    status_main = run_command_set(tox_env, "commands", chdir, ignore_errors, outcomes)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\session\cmd\run\single.py", line 96, in run_command_set
    current_outcome = tox_env.execute(
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\tox_env\api.py", line 362, in execute
    pass  # pragma: no cover
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\contextlib.py", line 126, in __exit__
    next(self.gen)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\tox_env\api.py", line 408, in execute_async
    self._execute_statuses.pop(execute_id)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\contextlib.py", line 126, in __exit__
    next(self.gen)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\tox_env\api.py", line 459, in _execute_call
    yield execute_status
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\contextlib.py", line 126, in __exit__
    next(self.gen)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\execute\api.py", line 136, in call
    out_sync.text,
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\tox\execute\stream.py", line 103, in text
    return self._content.decode("utf-8")
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe7 in position 146170: invalid continuation byte
.pkg: _exit> python C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta
  py39-test-alldeps: FAIL code 2 (711.19 seconds)
  evaluation failed :( (711.75 seconds)
Error: Process completed with exit code 2.
0s
```

**Comment by jugmac00** [comment]
Thanks for your report.

To get unblocked, you might decide to pin tox < 4.

That is quite possible a duplicate of https://github.com/tox-dev/tox/issues/2422

It would be super helpful if you take the time to create a minimal reproducer.

Also, as at least I have no access to a Windows box, some investigation of your would be highly appreciated.

**Comment by maxnoe** [comment]
> To get unblocked, you might decide to pin tox < 4.

Yes, this is already done and merged, we currently pin to 3.x.
